homeassistant/components/sensor/__init__.py

```diff
@@ -221,10 +221,6 @@ def unit_of_measurement(self) -> str | None:
     @property
     def state(self) -> Any:
```

Comment:
We should also mark the `state` method final and re-type `_attr_state: None = None`.